### PR TITLE
Team 2 decouple weapon

### PR DESCRIPTION
**Pull request by Magnius001**
# Description

Decoupling weapon from player entity by deleting the WeaponComponent and moving all necessary operation to the weapon itself.
- Now the WeaponFactory will handle the creation of the RangedWeapon/MeleeWeapon class, as well as the weapon entity for each weapon.
- The weapon will handle all operations by itself, the player entity will only need to call pickup, shoot, attack and drop.

Fixes / Closes # (issue)
#331 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] New unit tests for the creation of weapons, weapon entity, and components of weapon entity.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by SIN-Willett** [comment]
Overall I love this, the decoupling is awesome.

But I get a couple of issues when I run the game:
- [ ] File not found: images/screen/cutscene1.jpg... 
  - I reverted `IntroCutsceneDisplay.java` back to what it is in main and it fixed it.
- [ ] File not found: images/Weapons/pickaxe.atlas
  - From what I understand the pickaxe isn't animated, this I guess might be a problem

**Comment by Magnius001** [comment]
Issues sovled:
>     * [ ]  File not found: images/screen/cutscene1.jpg...
>       
- Reverted `IntroCutsceneDisplay.java` back to main.
> 
>     * [ ]  File not found: images/Weapons/pickaxe.atlas
>       
- Replaced `pickaxe.atlas` with `Axe.atlas`, replaced all references to "pickaxe" with "axe".

File: source/core/src/main/com/csse3200/game/components/player/inventory/ConcreteRangedWeapon.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+     * Create a projectile and launch it in the direction specified
+     * @param direction direction to shoot in
+     */
+    @Override
+    public void shoot(Vector2 direction) {
+        this.weaponEntity.getComponent(FiringController.class).activate(direction);
+    }
+
+    /**
+     * Reload the weapon
```
Comment (SIN-Willett) [inline source/core/src/main/com/csse3200/game/components/player/inventory/ConcreteRangedWeapon.java:124]: missing return.

File: source/core/src/main/com/csse3200/game/components/player/inventory/ConcreteRangedWeapon.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    public Entity getWeaponEntity() {
+        return this.weaponEntity;
+    }
+
+    public int getClipSize() {
```
Comment (SIN-Willett) [inline source/core/src/main/com/csse3200/game/components/player/inventory/ConcreteRangedWeapon.java:130]: missing comment.

File: source/core/src/main/com/csse3200/game/components/player/inventory/ConcreteRangedWeapon.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        this.iconPath = iconPath;
+        this.damage = damage;
+        this.range = range;
+        this.fireRate = fireRate;
+        this.ammo = ammo;
+        this.maxAmmo = maxAmmo;
+        this.reloadTime = reloadTime;
+    }
+
+    public void setWeaponEntity(Entity weaponEntity) {
```
Comment (SIN-Willett) [inline source/core/src/main/com/csse3200/game/components/player/inventory/ConcreteRangedWeapon.java:41]: missing comment

**Review by SIN-Willett — CHANGES_REQUESTED** [review]
Minor documentation request.

**Comment by Magnius001** [comment]
Added missing documentations

**Review by SIN-Willett — APPROVED** [review]

**Pull request merged**